utils/RESTWalker.py

```diff
@@ -0,0 +1,194 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import io
+import jinja2
+import json
+from pprint import pprint as pp
+import re
+
+import os
+
+UTILSDIR = os.path.abspath(os.path.dirname(__file__))
+COMMONPREFIX = os.path.dirname(UTILSDIR)
+SOURCE_ROOTDIR = os.path.join(COMMONPREFIX, 'f5', 'bigip')
+TEMPLATEDIR = os.path.join(UTILSDIR, 'template_library')
+DEVICECONFDIR = os.path.join(UTILSDIR, 'device_configs')
+
+
+class Kindless(Exception):
+    pass
+
+
+def load_json_from_pathname(pathname):
+    fp = io.open(pathname)
+    obj = json.load(fp)
+    fp.close()
+    return obj
+
+
+def dump_json(obj, pathname):
+    fp = io.open(pathname, 'wb')
+    json.dump(obj, fp, indent=1)
+    fp.close()
+
+
+class TemplateEngine(object):
+    def __init__(self, template_dir, config_dir):
+        self.OC_pattern =\
+            r'tm:(?P<OrgColl>\w+):(?P=OrgColl)collectionstate'
+        # Initialize templates
+        self.templates = {}
+        self._consume_directory(template_dir, self._template_consumer)
+        self.license_template = self.templates['license']
+        self.import_template = self.templates['imports']
+
+        # Initialize configs
+        self.raw_configs = {}
+        self.formatted_configs = {}
+        self._consume_directory(config_dir, self._config_consumer)
+
+    def _template_consumer(self, STUB, dirname, fnames):
+        for fname in fnames:
+            if fname.endswith('.tmpl'):
+                with io.open(os.path.join(dirname, fname)) as fh:
+                    templatekey = fname[:-len('.tmpl')]
+                    self.templates[templatekey] = jinja2.Template(fh.read())
+
+    def _config_consumer(self, STUB, dirname, fnames):
+        for fname in fnames:
+            if fname.endswith('.json'):
+                with io.open(os.path.join(dirname, fname)) as fh:
+                    configkey = fname[:-len('.json')]
+                    self.raw_configs[configkey] = json.load(fh)
+
+    def _consume_directory(self, dirname, consumer):
+        os.path.walk(dirname, consumer, self)
+
+    def list_templates(self):
+        pp(self.templates.keys())
+
+    def list_raw_configs(self):
+        pp(self.raw_configs.keys())
+
+    def process_config(self, config_name):
+        raw_conf = self.raw_configs[config_name]
+        if 'kind' in raw_conf:
+            python_as_string = self._process_config_with_kind(raw_conf)
+        else:
+            raise Kindless(raw_conf)
+        return python_as_string
+
+    def _klassify(self, raw_klass):
+        caps = '_'.join([x.capitalize() for x in raw_klass.split('-')])
+        dotless = '_'.join([x.capitalize() for x in caps.split('.')])
+        return dotless
+
+    def _build_import_dicts(self, raw_conf, klass):
+        imports = []
+        selfLinkstart = raw_conf[u"selfLink"].partition("?")[0]
```

Comment:
I propose we punt this:  https://github.com/F5Networks/f5-common-python/issues/223